source/PrintData.cpp

```diff
@@ -0,0 +1,497 @@
+/* PrintData.cpp
+Copyright (c) 2022 by warp-core
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#include "PrintData.h"
+
+#include "GameData.h"
+#include "Outfit.h"
+#include "Ship.h"
+
+#include <iostream>
+#include <map>
+#include <set>
+
+using namespace std;
+
+
+
+void PrintData::Print(const char *const *argv)
+{
+	const char *const *it = argv + 1;
+	if(!*it)
+		return;
+	string arg = *it;
+	if(arg == "-s" || arg == "--ships")
+		Ships(argv);
+	else if(arg == "-w" || arg == "--weapons")
+		PrintWeaponStats();
+	else if(arg == "-e" || arg == "--engines")
+		PrintEngineStats();
+	else if(arg == "--power")
+		PrintPowerStats();
+	else if(arg == "-o" || arg == "--outfits")
+		Outfits(argv);
+	cout.flush();
+}
+
+
+
+void PrintData::Help()
+{
+	cerr << "    -s, --ships: prints a table of ship stats (just the base stats, not considering any stored outfits)." << endl;
+	cerr << "    -s --sales: prints a table of ships with every 'shipyard' each appears in." << endl;
+	cerr << "    -s --loaded: prints a table of ship stats accounting for installed outfits. Does not include variants." << endl;
+	cerr << "    Use the modifier `-v` or `--variants` with the above command to include variants." << endl;
+	cerr << "    -w, --weapons: prints a table of weapon stats." << endl;
+	cerr << "    -e, --engines: prints a table of engine stats." << endl;
+	cerr << "    --power: prints a table of power outfit stats." << endl;
+	cerr << "    -o, --outfits: prints a list of outfits." << endl;
+	cerr << "    -o --sales: prints a list of outfits and every 'outfitter' each appears in." << endl;
+	cerr << "    -o -a, --all: prints a table of outfits and all attributes used by any outfits present." << endl;
+}
+
+
+
+void PrintData::Ships(const char *const *argv)
+{
+	bool loaded = false;
+	bool variants = false;
+	bool sales = false;
+
+	for(const char *const *it = argv + 2; *it; ++it)
+	{
+		string arg = *it;
+		if(arg == "-v" || arg == "--variants")
+			variants = true;
+		else if(arg == "--sales")
+			sales = true;
+		else if(arg == "--loaded")
+			loaded = true;
+	}
+
+	if(sales)
+		PrintShipShipyards();
+	else if(loaded)
+		PrintLoadedShipStats(variants);
+	else
+		PrintBaseShipStats();
```

Comment:
it doesn't support requesting multiple tables with a single command.
The reason it does a for loop on line 70 is so that it can check if the `variants` argument was included, because it needs to know that if it's printing the loaded ships table. It could in theory also be used with the base and sales tables, but such tables for the current vanilla game would provide no real additional info.